extensions/signal.sigmf-ext.md

```diff
@@ -0,0 +1,285 @@
+# The `signal` SigMF Extension Namespace v0.0.2
+
+## Description
+
+This document defines the `signal` extension namespace for the Signal
+Metadata Format (SigMF) specification. This extension namespace defines how to
+describe the attributes of wireless communications systems.
+
+## Copyright Notice
+
+This document is Copyright of The GNU Radio Foundation, Inc.
+
+This document is available under the [CC-BY-SA License](http://creativecommons.org/licenses/by-sa/4.0/).
+
+<a rel="license" href="http://creativecommons.org/licenses/by-sa/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" /></a>
+
+## Conventions Used in this Document
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+JSON keywords are used as defined in [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf).
+
+Augmented Backus-Naur form (ABNF) is used as defined by [RFC
+5234](https://tools.ietf.org/html/rfc5234) and updated by [RFC
+7405](https://tools.ietf.org/html/rfc7405).
+
+Fields defined as "human-readable", a "string", or simply as "text" shall be
+treated as plaintext where whitespace is significant, unless otherwise
+specified.
+
+## Specification
+
+This specification defines an extension `signal` namespace
+that can be used in SigMF `annotations` to describe wireless communications
+systems and their emitters.
+
+### Annotations
+
+This extension adds the following field to the `annotations` global SigMF object:
+
+|name|required|type|description|
+|----|--------------|-------|-----------|
+|`emission`|false|object|Describes the communications system of a signal emission and its emitter.|
+
+The field of communications is vast, and there may be communications systems that cannot be described using the names and fields described in this extension. If you need additional or different fields to describe a system, create a new extension that adds the necessary fields to the `signal` namespace and/or submit the new fields to be upstreamed into this specification.
+
+#### The `emission` Object
+
+`emission` objects contain name/value pairs that describe the modulations of communications systems. These objects MAY contain the following pairs:
+
+|name|required|type|description|
+|----|--------------|-------|----------|
+|`label`|false|string|A high-level shortform description useful to human readers (e.g., "wifi", "LTE")|
```

Comment:
I believe the `label` field is extremely common and should actually be moved to core (we could also call this `description` to be consistent with `global`), but this is really something I think should be up a level and more generic.

Reply:
This is an interesting point, @jacobagilbert. Currently, that's meant to be filled by the `comment` field, but I think what you're really highlighting is that it would be worthwhile to have a short-form human **and machine** -readable field that's in `core`. 

For purposes of process, I recommend we leave this field here, and then file a new issue & PR discussing and making that change across the codebase to keep the change atomic.

Reply:
Yes, thats right. 

Are you suggesting that we merge this here then separately move it to core after this is merged?